plugins/pencilblue/controllers/admin/plugins/plugin_settings.js

```diff
@@ -128,6 +135,36 @@ module.exports = function(pb) {
             });
         });
     };
+    function insertGroupHeadings(clone){
+        var previous = clone[clone.length - 1], current;
+        for (var i = clone.length - 1; i >= 0; i--) {
+            current = clone[i];
+            if (current.heading != previous.heading) {
+                clone.splice(i+1, 0, {id: previous.heading, isHeading: true, heading: convertNameToDisplayName(previous.heading)});
+            }
+
+            previous = current;
+        }
+    }
```

Comment:
This function doesn't seem to handle the case of displaying a header if the first setting has a group.  I'd like to propose the following:
```
    /**
     * Inserts a new record before the start of the new group.  This assumes that the settings array is already sorted 
     * by group
     * @param {Array} settings
     * @returns {Array}
     */
    PluginSettingsFormController.insertGroupHeadings = function(settings) {

        var prevHeading = '';
        var settingsAndHeaders = [];
        for (var i = 0; i < settings.length; i++) {
            if (prevHeading !== settings[i].heading) {
                settingsAndHeaders.push({
                    id: settings[i].heading,
                    isHeading: true,
                    heading: convertNameToDisplayName(settings[i].heading)
                });
            }
            prevHeading = settings[i].heading;
            settingsAndHeaders.push(settings[i]);
        }
        return settingsAndHeaders;
    };
```